packages/hoppscotch-app/helpers/curlparser.ts

```diff
@@ -2,105 +2,77 @@ import * as URL from "url"
+import { curlParserRequest } from "helpers/types/CurlParserResult"
 
 /**
  * given this: [ 'msg1=value1', 'msg2=value2' ]
  * output this: 'msg1=value1&msg2=value2'
  * @param dataArguments
  */
-const joinDataArguments = (dataArguments: string[]) => {
-  let data = ""
-  dataArguments.forEach((argument, i) => {
-    if (i === 0) {
-      data += argument
-    } else {
-      data += `&${argument}`
-    }
-  })
-  return data
-}
-
-const parseDataFromArguments = (parsedArguments: any) => {
-  if (parsedArguments.data) {
-    return {
-      data: Array.isArray(parsedArguments.data)
-        ? joinDataArguments(parsedArguments.data)
-        : parsedArguments.data,
-      dataArray: Array.isArray(parsedArguments.data)
-        ? parsedArguments.data
-        : null,
-      isDataBinary: false,
-    }
-  } else if (parsedArguments["data-binary"]) {
-    return {
-      data: Array.isArray(parsedArguments["data-binary"])
-        ? joinDataArguments(parsedArguments["data-binary"])
-        : parsedArguments["data-binary"],
-      dataArray: Array.isArray(parsedArguments["data-binary"])
-        ? parsedArguments["data-binary"]
-        : null,
-      isDataBinary: true,
-    }
-  } else if (parsedArguments.d) {
-    return {
-      data: Array.isArray(parsedArguments.d)
-        ? joinDataArguments(parsedArguments.d)
-        : parsedArguments.d,
-      dataArray: Array.isArray(parsedArguments.d) ? parsedArguments.d : null,
-      isDataBinary: false,
-    }
-  } else if (parsedArguments["data-ascii"]) {
-    return {
-      data: Array.isArray(parsedArguments["data-ascii"])
-        ? joinDataArguments(parsedArguments["data-ascii"])
-        : parsedArguments["data-ascii"],
-      dataArray: Array.isArray(parsedArguments["data-ascii"])
-        ? parsedArguments["data-ascii"]
-        : null,
-      isDataBinary: false,
-    }
-  }
-}
+const joinDataArguments = (dataArguments: string[]) => dataArguments.join("&")
 
 const parseCurlCommand = (curlCommand: string) => {
-  const newlineFound = /\\/gi.test(curlCommand)
-  if (newlineFound) {
-    // remove '\' and newlines
-    curlCommand = curlCommand.replace(/\\/gi, "")
-    curlCommand = curlCommand.replace(/\n/g, "")
-  }
+  // remove '\' and newlines
+  curlCommand = curlCommand.replace(/ ?\\ ?$/gm, " ")
+  curlCommand = curlCommand.replace(/\n/g, "")
+
+  // remove all $ symbols from start of argument values
+  curlCommand = curlCommand.replaceAll("$'", "'")
+  curlCommand = curlCommand.replaceAll('$"', '"')
+
   // replace string for insomnia
-  curlCommand = curlCommand.replace(/--request /, "-X ")
-  curlCommand = curlCommand.replace(/--header /, "-H ")
-  curlCommand = curlCommand.replace(/--url /, " ")
-  curlCommand = curlCommand.replace(/-d /, "--data ")
+  const replaceables: { [key: string]: string } = {
+    "--request": "-X",
+    "--header": "-H",
+    "--url": "",
+    "--form": "-F",
+    "--data-raw": "--data",
+    "--data": "-d",
+    "--user": "-u",
+  }
```

Comment:
I don't think the explicit type definition is necessary ?

Also, I think we can move this variable outside the function as this is a constant and we don't need to create the same thing all the time.

Reply:
> I don't think the explicit type definition is necessary ?

It creates these type errors:

```
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type
'{ "--request": string; "--header": string; "--url": string; "--form": string; "--data-raw": string; "--data": string; "--user": string; }'.

No index signature with a parameter of type 'string' was found on type
'{ "--request": string; "--header": string; "--url": string; "--form": string; "--data-raw": string; "--data": string; "--user": string; }'.ts(7053)
```

Even though `r` is taken from replaceables
```ts
  for (const r in replaceables) {
    for (const sym of ["'", '"', " "])
      curlCommand = curlCommand.replace(RegExp(r + sym), replaceables[r] + sym)
  }
```